### PR TITLE
Pll retune

### PR DESCRIPTION
**Pull request by henryhallam**
Work in progress; needs testing; needs updates to libswiftnav-python bindings

### CODE REVIEW

**Comment by henryhallam** [comment]
Goes with swift-nav/piksi_firmware#456 @fnoble @gsmcmullin 
libswiftnav-python bindings have been updated now also.

File: src/pvt.c [hunk 1]
```diff
@@ -271,8 +271,9 @@ static u8 filter_solution(gnss_solution* soln, dops_t* dops)
    * regulations. It must not be removed. Any modification to this condition
    * is strictly not approved by Swift Navigation Inc. */
 
-  if (vector_norm(3, soln->vel_ecef) >= 0.514444444*1000)
-    /* Velocity is greater than 1000kts. */
+  if ((vector_norm(3, soln->vel_ecef) >= 0.514444444*1000)
+      && (soln->pos_llh[2] > 18000))
+    /* COCOM limits: velocity greater than 1000 kts and altitude > 18 km. */
```
Comment (fnoble) [inline src/pvt.c]: Please revert this, COCOM does state altitude > 18km but DOC states just 1000kts (no minimum altitude).

File: src/track.c [hunk 2]
```diff
@@ -339,14 +357,21 @@ void aided_tl_init(aided_tl_state_t *s, float loop_freq,
...
+    s->prev_Q = cs[1].Q;
+  }
   s->carr_freq = aided_lf_update(&(s->carr_filt), carr_error, freq_error);
 
+  /* Code loop */
   float code_error = dll_discriminator(cs);
-  s->code_freq = simple_lf_update(&(s->code_filt), -code_error); // + s->carr_freq * SCALING_FACTOR
+  s->code_freq = simple_lf_update(&(s->code_filt), -code_error);
+  if (s->carr_to_code) /* Optional carrier aiding of code loop */
+    s->code_freq += (s->carr_freq) / s->carr_to_code;
```
Comment (fnoble) [inline src/track.c]: parens needed?

**Comment by fnoble** [comment]
lgtm after minor fixes

File: src/pvt.c (hunk 1 above)
Comment (henryhallam) [inline src/pvt.c]: Hmm, per ECCN 7A105 there is indeed a pure velocity limit but it's actually 600 m/s not 1000 knots.  The 1000 knots / 18 km language has been deleted from the ITAR algoteher.  Anyway it's not related to this PR, I'll pull it out.

Comment (fnoble) [inline src/pvt.c]: Yeah seeing as 600m/s is just a little over 1000kts I thought putting a 1000kt limit would satisfy both requirements in a simple way.

**Comment by henryhallam** [comment]
This PR contains:
- Retunable loop filters, now with optional carrier aiding
- Max-likelihood nav bit synchronization
- Explicit half-cycle slip detection
- Improved C/N0 estimator

Closes #173 #424

**Pull request merged**